### PR TITLE
doc: update readme

### PR DESCRIPTION
**Pull request by jacten**
Updated documentation a bit. Nothing crazy as I am lazy but just more up to date.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,208 +10,48 @@ Another problem, specific to video-conferencing is it was difficult for the acto
...
 
 ```
-games[games.length - 1].endTime !== null;
+MONGO_URI="mongodb+srv://<username>:<password>@cluster0-cusns.mongodb.net/<database>?retryWrites=true&w=majority"
 ```
 
-## Timer Thoughts
-
-Simple solution would be to set the start time for a turn as a future moment. And then use `setInterval` on client with a countdown 3 sec before turn `startTime` and then counting down to 0 based on the `turnDurationSeconds` setting.
+I put an example `.env` file in `.env.sample` feel free to `mv .env.sample .env` to get started.
```
Comment (jgsheppa) [inline README.md:48]: I like the new explanation of the MongoDB set up. Nice job!

File: documentation/Timer.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+### Problem
+
+We knew we wanted to start a timer at the same time for all users across different devices in different timezones. We also knew that we wanted this to be a mostly client side process where once the timer starts it does not need to keep pinging the server to check if it is still at the same time. Ultimately the times didn't need to be 100% accurate with each other but should be close enough for no confusion.
+
+### First Approach - setInterval()
+
+When the actor clicks "ready" the server would set the round start as 5 seconds in the future and emit that updated party object to all clients. Then client side the timer would start when their system time was equal to the remote time. The timer would then use `setInterval` to increment the timer each second.
+
+Unfortunatly `setInterval` is unreliable. If a users computer or phone is under heavy load, the interval will fire late. Similarly, if the users phone goes to sleep or navigates away from the browser, the interval will pause until they refocus on the window. This meant that users would see wildly different times.
```
Comment (jgsheppa) [inline documentation/Timer.md]: Unfortunately is missing an 'e.'

File: documentation/Timer.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Another potential problem with this, is that if a users system clock happens to be different from our server (something rare in these modern times but still worth considering) then the timers would be off by that difference.
+
+These three problems: unreliable intervals, falling asleep phones, and differing system clocks meant we needed a new solution.
+
+### Working Solution - setTimeout() with resync at every 10th of a second with server/client system clock sync
+
+To solve the issue of differing system clocks we calulate the difference between the servers time and each client and use add that difference to the end time of a round (see `useServerTime.js` hook). This means that even if a client clock had been manually set to something completely random the timer should still be in sync.
+
+Two fix the issues caused by setInterval, we use recursivly call setTimeout every 10th of a second until the time runs out. Every 10th a second we re-calculate the time remaining and update the countdown and percentage.
```
Comment (jgsheppa) [inline documentation/Timer.md]: 'To' instead of 'Two.'

Comment (jgsheppa) [inline documentation/Timer.md]: Every 10th **of** a second...

**Review by jgsheppa — CHANGES_REQUESTED** [review]
I added a few comments where I saw small typos. Little stuff, but thought I should let you know.

File: documentation/Timer.md (hunk 3 above)
Comment (chedgo) [reply documentation/Timer.md]: recursivly

File: documentation/Timer.md (hunk 2 above)
Comment (jacten) [reply documentation/Timer.md]: how unfortunate

**Comment by jacten** [comment]
Thanks for fixes @chedgo  and @jgsheppa . I'll just merge since i'm admin and didn't do anything besides fix grammar / spelling.

**Pull request merged**